src/components/RegistrationForm.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState, ChangeEventHandler, ChangeEvent } from "react";
+import { db as firebaseDB } from "../services/firebase-client";
+import { sha256 } from "js-sha256";
+
+export const RegistrationForm = ({ onRegistrationDone = (() => {}) }) => {
+  
+  const [username, setUsername] = useState("");
+  const [redemptionType, setRedemptionType] = useState("");
+
+  const InputChangeHandlers: { [key: string]: ChangeEventHandler<HTMLInputElement> } = {
+    username: (e: ChangeEvent<HTMLInputElement>) => {
+      setUsername(e.target.value);
+    },
+    redemptionType: (e: ChangeEvent<HTMLInputElement>) => {
+      setRedemptionType(e.target.value);
+    },
+  };
+  const createInputChangeHandler = ( name: string ): ChangeEventHandler<HTMLInputElement> => {
+    const handleInputChange = InputChangeHandlers[name] || ((event: ChangeEvent<HTMLInputElement>) => {});
+
+    return handleInputChange;
+  };
+  const registerUser = async () => {
+    const createDate = new Date();
+    const redemptionString = `${username}_${redemptionType}-${createDate}`;
+    const redemptionCode = sha256(redemptionString)
+
+    try {
+      await firebaseDB.append("users", {
+        username,
+        redemptionType,
+        redemptionCode,
+        createDate,
+      });
```

Comment:
Its bad practice to invoke db call in Representation layer (UI components)

I expect the `registerUser` function should invoke a "service"/ "network" module, i.e.

```typescript
  const registerUser = async () => {
    await service.registerUser({username, redemptionType)
  };
```

```typescript
# service.tsx

const registerUser = (userForm = {username: string, redemptionType: string}) => {
    string redemptionCode = _buildRedemptionCode(username, redemptionType)
    # firebase thing
    return user
}

const _buildRedemptionCode = (username: string, redemptionType: string) => {
    const createDate = new Date();
    const redemptionString = `${username}_${redemptionType}-${createDate}`;
    const redemptionCode = sha256(redemptionString)
    return redemptionCode
}
```